ecosystem/gstreamer_plugin/gst_mtl_st20p_tx.c

```diff
@@ -158,137 +149,62 @@ static void gst_mtl_st20p_tx_class_init(Gst_Mtl_St20p_TxClass* klass) {
+      gobject_class, PROP_ST20P_TX_RETRY,
+      g_param_spec_uint("retry", "Retry Count",
+                        "Number of times the MTL will try to get a frame.", 0, G_MAXUINT,
+                        10, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
 
   g_object_class_install_property(
-      gobject_class, PROP_TX_FRAMERATE,
+      gobject_class, PROP_ST20P_TX_FRAMERATE,
       g_param_spec_uint("tx-fps", "Video framerate", "Framerate of the video.", 0,
                         G_MAXUINT, 0, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
 
   g_object_class_install_property(
-      gobject_class, PROP_TX_FRAMEBUFF_NUM,
+      gobject_class, PROP_ST20P_TX_FRAMEBUFF_NUM,
       g_param_spec_uint("tx-framebuff-num", "Number of framebuffers",
                         "Number of framebuffers to be used for transmission.", 0,
                         G_MAXUINT, 3, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
 }
 
 static gboolean gst_mtl_st20p_tx_start(GstBaseSink* bsink) {
   struct mtl_init_params mtl_init_params = {0};
-  gint ret;
 
   Gst_Mtl_St20p_Tx* sink = GST_MTL_ST20P_TX(bsink);
 
   GST_DEBUG_OBJECT(sink, "start");
   GST_DEBUG("Media Transport Initialization start");
   gst_base_sink_set_async_enabled(bsink, FALSE);
 
-  /* mtl is already initialzied */
-  if (sink->mtl_lib_handle) {
-    GST_INFO("Mtl already initialized");
-    if (mtl_start(sink->mtl_lib_handle)) {
-      GST_ERROR("Failed to start MTL");
+  /* Check for future support of multiple plugins initialized from the same handle.
+   * TODO: Add support for initializing multiple pipelines from a single handle. */
+  if (!sink->mtl_lib_handle) {
+    if (!gst_mtl_common_parse_dev_arguments(&mtl_init_params, &(sink->devArgs))) {
+      GST_ERROR("Failed to parse dev arguments");
       return FALSE;
     }
-    return TRUE;
-  }
-
-  strncpy(mtl_init_params.port[MTL_PORT_P], sink->devArgs.port, MTL_PORT_MAX_LEN);
-
-  ret = inet_pton(AF_INET, sink->devArgs.local_ip_string,
-                  mtl_init_params.sip_addr[MTL_PORT_P]);
-  if (ret != 1) {
-    GST_ERROR("%s, sip %s is not valid ip address\n", __func__,
-              sink->devArgs.local_ip_string);
-    return false;
-  }
-
-  if (sink->devArgs.tx_queues_cnt[MTL_PORT_P]) {
-    mtl_init_params.tx_queues_cnt[MTL_PORT_P] = sink->devArgs.tx_queues_cnt[MTL_PORT_P];
-  } else {
-    mtl_init_params.tx_queues_cnt[MTL_PORT_P] = 16;
-  }
-
-  mtl_init_params.rx_queues_cnt[MTL_PORT_P] = 0;
-  mtl_init_params.num_ports++;
-
-  mtl_init_params.flags |= MTL_FLAG_BIND_NUMA;
-  if (sink->silent) {
-    mtl_init_params.log_level = MTL_LOG_LEVEL_ERROR;
-  } else {
-    mtl_init_params.log_level = MTL_LOG_LEVEL_INFO;
-  }
-
-  sink->retry_frame = 10;
-
-  mtl_init_params.flags |= MTL_FLAG_BIND_NUMA;
 
-  if (sink->devArgs.dma_dev) {
-    strncpy(mtl_init_params.dma_dev_port[0], sink->devArgs.dma_dev, MTL_PORT_MAX_LEN);
-  }
+    if (sink->log_level && sink->log_level < MTL_LOG_LEVEL_MAX) {
```

Comment:
why MTL_LOG_LEVEL_DEBUG ( which is 0 ) is not allowed?

Reply:
also, I would pass `sink->log_level` to `gst_mtl_common_parse_dev_arguments()` and set log level there instead to in every plugin

Reply:
Ok fixed